pkg/apis/config/v1beta1/types.go

```diff
@@ -130,13 +130,32 @@ type LoadVariationRiskBalancingArgs struct {
 	SafeVarianceSensitivity *float64 `json:"safeVarianceSensitivity,omitempty"`
 }
 
+// ScoringStrategyType is a "string" type.
+type ScoringStrategyType string
+
+const (
+	// MostAllocated strategy favors node with the least amount of available resource
+	MostAllocated ScoringStrategyType = "MostAllocated"
+	// BalancedAllocation strategy favors nodes with balanced resource usage rate
+	BalancedAllocation ScoringStrategyType = "BalancedAllocation"
+	// LeastAllocated strategy favors node with the most amount of available resource
+	LeastAllocated ScoringStrategyType = "LeastAllocated"
+)
+
+type ScoringStrategy struct {
+	Type      ScoringStrategyType            `json:"type,omitempty"`
+	Resources []schedulerconfig.ResourceSpec `json:"resources,omitempty"`
+}
+
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:defaulter-gen=true
 
 // NodeResourceTopologyMatchArgs holds arguments used to configure the NodeResourceTopologyMatch plugin
 type NodeResourceTopologyMatchArgs struct {
 	metav1.TypeMeta `json:",inline"`
 
-	KubeConfigPath *string  `json:"kubeconfigpath,omitempty"`
-	MasterOverride *string  `json:"masteroverride,omitempty"`
-	Namespaces     []string `json:"namespaces,omitempty"`
+	KubeConfigPath  *string         `json:"kubeconfigpath,omitempty"`
+	MasterOverride  *string         `json:"masteroverride,omitempty"`
+	Namespaces      []string        `json:"namespaces,omitempty"`
+	ScoringStrategy ScoringStrategy `json:"scoringStrategy,omitempty"`
```

Comment:
```suggestion
	ScoringStrategy *ScoringStrategy `json:"scoringStrategy,omitempty"`
```

As it's the external versioned type.

Reply:
Again thank you for taking the time and review everything.
But after this change, running ```hack/update-codegen.sh``` ended up with:
```
E0820 08:11:19.961016 2283072 conversion.go:755] Warning: could not find nor generate a final Conversion function for sigs.k8s.io/scheduler-plugins/pkg/apis/config/v1beta1.NodeResourceTopologyMatchArgs -> sigs.k8s.io/scheduler-plugins/pkg/apis/config.NodeResourceTopologyMatchArgs
E0820 08:11:19.961063 2283072 conversion.go:756]   the following fields need manual conversion:
E0820 08:11:19.961067 2283072 conversion.go:758]       - ScoringStrategy
E0820 08:11:19.961338 2283072 conversion.go:755] Warning: could not find nor generate a final Conversion function for sigs.k8s.io/scheduler-plugins/pkg/apis/config.NodeResourceTopologyMatchArgs -> sigs.k8s.io/scheduler-plugins/pkg/apis/config/v1beta1.NodeResourceTopologyMatchArgs
E0820 08:11:19.961351 2283072 conversion.go:756]   the following fields need manual conversion:
E0820 08:11:19.961356 2283072 conversion.go:758]       - ScoringStrategy
```
Maybe it's an additional bug in the code generator?

Reply:
I _think_ it's as expected. The defaulter logic will be able to cover it. Could you add a UT in cmd/scheduler/main_test.go to ensure an argument without specifying the scoringStrategy field can be instantiated properly?

BTW: the codegen script somehow removes two functions `Convert_v1beta1_NodeResourceTopologyMatchArgs_To_config_NodeResourceTopologyMatchArgs` and `Convert_config_NodeResourceTopologyMatchArgs_To_v1beta1_NodeResourceTopologyMatchArgs`, which causes a compilation error. Not quite sure it's a bug. I tried to add them back and it looks good. FYI: https://github.com/Huang-Wei/scheduler-plugins/commit/a91ae79ae578f63462aecc45da1b1634a9adf190.

Reply:
> I _think_ it's as expected. The defaulter logic will be able to cover it. Could you add a UT in cmd/scheduler/main_test.go to ensure an argument without specifying the scoringStrategy field can be instantiated properly?

Without adding logic for manual conversion of scoringStrategy field in https://github.com/kubernetes-sigs/scheduler-plugins/pull/220/commits/563a9ed8e2d437aa858cfcc9fddd4b7665de6fc7#diff-9c6b4d879381c71601a34af877d2fa07eb444b866564a0e9e6050e97bcfd4557R274 the UT  in cmd/scheduler/main_test.go are falling. 
I tried stick into the conversion as it shown in [noderesourcefit](https://github.com/kubernetes/kubernetes/blob/b9565beef015df9ae8e73b408d106531bea5ba77/pkg/scheduler/apis/config/v1beta2/zz_generated.conversion.go#L515)
the only difference is that here the conversion is concrete <-> pointer